src/main/java/org/kairosdb/core/datastore/DataPointRowImpl.java

```diff
@@ -21,7 +21,7 @@
 
 public class DataPointRowImpl implements DataPointRow
 {
-	private List<DataPoint> dataPoints = new ArrayList<DataPoint>();
+	private List<DataPoint> dataPoints = new LinkedList<DataPoint>();
```

Comment:
Nitpick:

Why change implementations here?  Do we know how this list is used in all cases?  This raises to the complexity of get to O(n).  While add is complexity of 1 for prepending, ArrayList also has complexity of one for appending.

If we generally know the sizes of the data points array on creation we can also ensure no resizing.

Reply:
Looking through the code a bit, it seems there are a lot of places where we are one-by-one allocating converted-for-kairos datapoints when we already know the size of the entire batch.  See InfluxDBDatastore:157 for an example.

We can allocate an adequate array and use it to back an ArrayList a single time for each series result in one go instead of one-by-one, and take full advantage of the perf benefits of ArrayList over LinkedList.  This would require changing the QueryCallback interface, but we have full control over it.

Worth considering as this behavior is in all the performance critical pathways.